examples/cert-validation.cpp

```diff
@@ -0,0 +1,248 @@
+            exit(EXIT_FAILURE);
+        }
+        std::cout << "Certificate successfully validated\n---\n";
+    }
+
+    /*
+        Basic certificate validation using X509Certificate::verify() against
+        issuing certificate chain
+    */
+    {
+        std::cout << "Validating certificate against certificate chain\n";
+        try {
+            cert.verify({rootCA}, {intermediateCA});
+        } catch (const MoCOCrWException &e) {
+            std::cerr << "Failed to validate certificate.\n";
+            std::cerr << "OpenSSL error: " << e.what() << '\n';
+            exit(EXIT_FAILURE);
+        }
+        std::cout << "Certificate successfully validated\n---\n";
+    }
+
+    /*
+        Basic positive case - CA -> cert validation using VerificationContext class.
+        VerificationContext is a class that can be used to define the "environment"
+        in which x509 certificates are validated. This makes it easy to setup custom
+        validation rules upfront and then validate the certificates according to the
+        rules.
+
+        If validation fails, MoCOCrW exception is thrown that contains error
+        message from the underlying openssl library.
+    */
+    std::cout << "\n*** Certificate validation examples using VerificationContext ***\n\n";
+    {
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificate(intermediateCA);
+        std::cout << "Validating certificate against an issuing CA\n";
+        try {
+            cert.verify(ctx);
+        } catch (const MoCOCrWException &e) {
+            std::cerr << "Failed to validate certificate.\n";
+            std::cerr << "OpenSSL error: " << e.what() << '\n';
+            exit(EXIT_FAILURE);
+        }
+        std::cout << "Certificate successfully validated\n---\n";
+    }
+
+    /*
+        Basic positive case - cert validation within certificate chain using
+        VerificationContext
+    */
+    {
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificate(rootCA)
+           .addIntermediateCertificate(intermediateCA);
+        std::cout << "Validating certificate against certificate chain\n";
+        try {
+            cert.verify(ctx);
+        } catch (const MoCOCrWException &e) {
+            std::cerr << "Failed to validate certificate.\n";
+            std::cerr << "OpenSSL error: " << e.what() << '\n';
+            exit(EXIT_FAILURE);
+        }
+        std::cout << "Certificate inside certificate chain successfully validated\n---\n";
+    }
+
+    /*
+        Negative case - Validation attempt of the certificate issued by other CA's.
+    */
+    {
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificate(rootCA)
+           .addIntermediateCertificate(intermediateCA);
+        std::cout << "Verifying certificate that is issued from other CA's\n";
+        try {
+            cert_from_other_chain.verify(ctx);
```

Comment:
We should probably exit with an error after this line as this shouldn't be reachable if everything works as expected.